components/director/internal/domain/application/service.go

```diff
@@ -442,40 +432,67 @@ func (s *service) createRelatedResources(ctx context.Context, in model.Applicati
 		return errors.Wrapf(err, "while creating Webhooks for application")
 	}
 
-	for _, item := range in.Apis {
+	err = s.createApis(ctx, applicationID, tenant, in.Apis)
+	if err != nil {
+		return errors.Wrapf(err, "while creating APIs for application")
+	}
+
+	err = s.createEvents(ctx, applicationID, tenant, in.EventAPIs)
+	if err != nil {
+		return errors.Wrapf(err, "while creating Events for application")
+	}
+
+	err = s.createDocuments(ctx, applicationID, tenant, in.Documents)
+	if err != nil {
+		return errors.Wrapf(err, "while creating Documents for application")
+	}
+
+	return nil
+}
+
+func (s *service) createApis(ctx context.Context, appID, tenant string, apis []*model.APIDefinitionInput) error {
```

Comment:
I think it would be nice to keep the name consistent across the project and capitilize it into `API`